pandas/core/ops.py

```diff
@@ -754,7 +754,8 @@ def wrapper(self, other, axis=None):
         elif isinstance(other, pd.DataFrame):  # pragma: no cover
             return NotImplemented
         elif isinstance(other, (np.ndarray, pd.Index)):
-            if len(self) != len(other):
+            lo = len(np.atleast_1d(other))
+            if lo != 1 and len(self) != lo:
```

Comment:
why are you adding this check for != 1? this seems to be hiding thing. Further why don't you do:

``other = np.atleast_1d(other)`` and be done?